clients/src/main/java/org/apache/kafka/common/requests/EnvelopeRequest.java

```diff
@@ -91,4 +91,14 @@ public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
     public static EnvelopeRequest parse(ByteBuffer buffer, short version) {
         return new EnvelopeRequest(ApiKeys.ENVELOPE.parseRequest(version, buffer), version);
     }
+
+    public EnvelopeRequestData data() {
+        return data;
+    }
+
+    @Override
+    public Send toSend(String destination, RequestHeader header) {
```

Comment:
If all requests are using auto-generated data, should this be default implementation of ```AbstractRequest```?

Reply:
Yeah, I think so. And looks like we're almost there. After your patch for `Produce`, the only remaining unconverted API that I see is `OffsetsForLeaderEpoch`.

Reply:
#7409 might be a good opportunity to complete this. What do you think @ijuma ?